src/lib/comms/include/sol-coap-transport.h

```diff
@@ -0,0 +1,197 @@
+    int (*sendmsg)(struct sol_coap_transport *transport,
+        const void *buf, size_t len, const struct sol_network_link_addr *addr);
+
+    /**
+     * @brief Receives data
+     *
+     * @param transport The pointer to this struct (self).
+     * @param buf The buffer that will be used to store the received data.
+     * @param len The size of the given data's buffer.
+     * @param addr The address from where the data came.
+     * @return 0 on success, -errno on error.
+     */
+    int (*recvmsg)(struct sol_coap_transport *transport,
+        void *buf, size_t len, struct sol_network_link_addr *addr);
+
+    /**
+     * @brief Set a callback to be called when is possible to send data.
+     *
+     * @param transport The pointer to this struct (self).
+     * @param on_can_write The callback to be called.
+     * @param user_data The data that will be given when the callback be called.
+     * @return 0 on success, -errno on error.
+     */
+    int (*set_on_write)(struct sol_coap_transport *transport,
+        bool (*on_can_write)(void *data, struct sol_coap_transport *transport),
+        const void *user_data);
+
+    /**
+     * @brief Set a callback to be called when there is available data to read.
+     *
+     * @param transport The pointer to this struct (self).
+     * @param on_can_read The callback to be called.
+     * @param user_data The data that will be given when the callback be called.
+     * @return 0 on success, -errno on error.
+     */
+    int (*set_on_read)(struct sol_coap_transport *transport,
+        bool (*on_can_read)(void *data, struct sol_coap_transport *transport),
+        const void *user_data);
+};
+
+/**
+ * @brief Sends data through a CoAP transport.
+ *
+ * This function is wrapper over the @c sendmsg method of the @c sol_coap_transport
+ * struct. It checks the parameters and the API version of the transport.
+ *
+ * @param transport The pointer to this struct (self).
+ * @param buf The data that is wanted to be sent.
+ * @param len The size of the given data's buffer.
+ * @param addr The address that will be sent.
+ * @return 0 on success, -errno on error.
+ *
+ * @see sol_coap_transport_set_on_write()
+ */
+int sol_coap_transport_sendmsg(struct sol_coap_transport *transport, const void *buf, size_t len,
+    const struct sol_network_link_addr *addr);
+
+/**
+ * @brief Receives data using from a CoAP transport.
+ *
+ * This function is wrapper over the @c recvmsg method of the @c sol_coap_transport
+ * struct. It checks the parameters and the API version of the transport.
+ *
+ * @param transport The pointer to this struct (self).
+ * @param buf The buffer that will be used to store the received data.
+ * @param len The size of the given data's buffer.
+ * @param addr The address from where the data came.
+ * @return 0 on success, -errno on error.
+ *
+ * @see sol_coap_transport_set_on_read()
+ */
+int sol_coap_transport_recvmsg(struct sol_coap_transport *transport, void *buf, size_t len,
```

Comment:
return looks strange, you should match `sendmsg()` that returns `ssize_t` and thus the amount received, not 0 on success.